File: CloudMusic/src/main/java/com/ken/android/CloudMusic/Service/Service.java
package com.ken.android.CloudMusic.Service;

import android.app.Activity;
import android.app.ActivityManager;
import android.content.Context;

import java.util.List;

/**
 * Created by axnshy on 16/7/30.
 */
public class Service{
    /*
    *判断service有没有运行
    * */
    public static boolean isMyServiceRunning(Context context,Class<?> serviceClass) {
        ActivityManager manager = (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);
        for (ActivityManager.RunningServiceInfo service : manager.getRunningServices(Integer.MAX_VALUE)) {
            if (serviceClass.getName().equals(service.service.getClassName())) {
                return true;
            }
        }
        return false;
    }
}

File: SmartPlay/src/main/java/com/luffy/smartplay/player/Service.kt
package com.luffy.smartplay.player

import android.app.ActivityManager
import android.content.*

/**
 * Created by axnshy on 16/7/30.
 */
object Service {
    /*
    *判断service有没有运行
    * */
    fun isMyServiceRunning(context: Context?, serviceClass: Class<*>): Boolean {
        val manager = context!!.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
        for (service in manager.getRunningServices(Int.MAX_VALUE)) {
            if (serviceClass.name == service.service.className) {
                return true
            }
        }
        return false
    }
}